internal/blobstore/multistore.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package blobstore // import "gopkg.in/juju/charmstore.v5-unstable/internal/blobstore"
+
+import (
+	"io"
+
+	"gopkg.in/errgo.v1"
+)
+
+// multiStore is a store.
+var _ store = (*multiStore)(nil)
+
+// multiStore holds the Stores to use as the blobstore.
+type multiStore struct {
+	stores []*Store
+	// paralle arrays because i'm sloppy
```

Comment:
How about making the type and mode part of the Store interface?
Then you won't need the parallel slices.